app/src/main/java/org/septa/android/app/transitview/TransitViewResultsActivity.java

```diff
@@ -0,0 +1,574 @@
+                                } catch (Exception e1) {
+                                    Log.e(TAG, "Failed to move camera, defaulting map zoom to Philadelphia City Hall");
+
+                                    // TODO: where should I move map to on default?
+                                    googleMap.moveCamera(CameraUpdateFactory.zoomTo(13));
+                                    googleMap.moveCamera(CameraUpdateFactory.newLatLng(cityHall));
+                                }
+                                // TODO: calculate map height
+//                                ViewGroup.LayoutParams layoutParams =
+//                                        bottomSheetLayout.getLayoutParams();
+//                                layoutParams.height = rootView.getHeight() - mapContainerView.getTop();
+//                                bottomSheetLayout.setLayoutParams(layoutParams);
+//                                bottomSheetBehavior.setPeekHeight(peekHeight);
+                            }
+                        });
+                    }
+                }
+
+                TransitViewVehicleDetailsInfoWindowAdapter adapter = new TransitViewVehicleDetailsInfoWindowAdapter(TransitViewResultsActivity.this);
+                googleMap.setInfoWindowAdapter(adapter);
+            }
+        });
+
+        // hide progress view and show map
+        progressView.setVisibility(View.GONE);
+        mapContainerView.setVisibility(View.VISIBLE);
+//        alertsView.setVisibility(View.VISIBLE); // TODO: show alerts
+
+        int permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION);
+        if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
+            Log.d(TAG, "Location Permission Granted.");
+            Task<Location> locationTask = LocationServices.getFusedLocationProviderClient(this).getLastLocation()
+                    .addOnSuccessListener(this, new OnSuccessListener<Location>() {
+                        @Override
+                        public void onSuccess(Location location) {
+                            // Got last known location. In some rare situations this can be null.
+                            if (location != null) {
+                                int permissionCheck = ContextCompat.checkSelfPermission(TransitViewResultsActivity.this,
+                                        Manifest.permission.ACCESS_FINE_LOCATION);
+                                googleMap.setMyLocationEnabled(true);
+                            } else {
+                                Log.d(TAG, "location was null");
+                            }
+                        }
+                    });
+        }
+    }
+
+    @Override
+    public boolean isTrolley(String routeId) {
+        String[] trolleyRouteIds = new String[]{"10", "11", "13", "15", "34", "36", "101", "102"};
```

Comment:
If you made this part of res/values you wouldn't have to duplicate your code.  Now if SEPTA adds a trolly route they will have to change it in at least 2 places.

Reply:
I have it as a single static method in a utils class in the 85 branch, which has the rest of the features implemented.